restore-sql-server-transaction-logs/Function/main.py

```diff
@@ -0,0 +1,258 @@
+            blob_metadata["BackupType"].lower() not in ("full", "diff", "tlog")):
+        raise RuntimeError("The tag BackupType must be defined and must take the values FULL, DIFF or TLOG")    
+        
+    if ("Recovery" not in (blob_metadata) or not blob_metadata["Recovery"] or
+            blob_metadata["Recovery"].lower() not in ("true", "false")):
+        raise RuntimeError("The tag Recovery must be defined and must take the values True or False")
+    
+    csql_instance_name = blob_metadata["CloudSqlInstance"]
+    database_name = blob_metadata["DatabaseName"]
+    backup_type = blob_metadata["BackupType"].upper()
+    no_recovery = not(blob_metadata["Recovery"].lower() == "true")
+
+    return csql_instance_name, database_name, backup_type, no_recovery    
+
+
+def get_import_context(source_bucket_name: str, object_name: str, database_name: str, backup_type: str, no_recovery:str) -> dict:
+    
+    uri="gs://"+source_bucket_name+"/"+object_name
+
+    instances_import_context = {
+        "importContext": {
+            "fileType": "BAK",
+            "uri": uri,
+            "database": database_name,
+            "bakImportOptions": {
+                    "bakType": backup_type,
+                    "noRecovery": no_recovery,
+                }
+        }
+    }
+
+    return instances_import_context
+
+
+def delete_processed_blob(source_bucket: google.cloud.storage.bucket.Bucket,object_name: str, logger) -> None:
+    try:
+        source_bucket.delete_blob(object_name)
+        logger.info(f"Deleted object {object_name} from the source bucket.")
+    except Exception as err:
+        raise RuntimeError(f"Could not delete the object {object_name} from the source bucket. Error: {err.args}")
+    return
+
+
+def handle_error(error_response: dict, source_bucket: google.cloud.storage.bucket.Bucket, object_name: str, logger) -> [str, bool]:
+    logger.info(f"Got an error in the operation response: {error_response}")
+
+    #if the import fails with error Msg 4326 - too early to apply to the database
+    #delete the object assuming it was already processed earlier
+    if "Msg 4326" in str(error_response.get("errors")):
+        delete_processed_blob(source_bucket, object_name, logger)
+        logger.info(f"Got a too early to apply to the database. Finished processing object {object_name}")
+        return "Operation succeded.", True
+
+    #if the import fails with error Msg 4305 - too recent to apply to the database
+    #leave the object on the bucket showing that it was not processed.
+    elif "Msg 4326" in str(error_response.get("errors")):
+        return f"Operation failed. Got a too recent SQL error and did not process the object {object_name}", False
+
+    #if the import fails for any other reason, display the inner error stack.
+    else:
+        return f"Operation failed. Could not process the object {object_name}. Error: {error_response}.", False

```

Comment:
Are these error conditions? If so, we should throw an exception instead

Reply:
Agreed. Done.

Reply:
LGTM